File: leetcode_java/88. Merge Sorted Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int tail = m + n - 1, i = m - 1, j = n - 1;
        while (i >= 0 && j >= 0) {
            if (nums1[i] > nums2[j]) nums1[tail] = nums1[i--];
            else nums1[tail] = nums2[j--];
            tail--;
        }
        for(int k = j; k >= 0; k--) nums1[k] = nums2[k];
    }
}

File: leetcode_python/88. Merge Sorted Array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        tail, i, j = m + n - 1, m - 1, n - 1
        while i >= 0 and j >= 0:
            if nums1[i] > nums2[j]:
                nums1[tail] = nums1[i]
                i -= 1
            else:
                nums1[tail] = nums2[j]
                j -= 1
            tail -= 1
        for k in range(j, -1, -1): nums1[k] = nums2[k]